clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -142,7 +143,7 @@ public MemoryRecordsBuilder(ByteBuffer buffer,
         this.initialCapacity = buffer.capacity();
 
         if (magic > LogEntry.MAGIC_VALUE_V1) {
-            buffer.position(initPos + EosLogEntry.RECORDS_OFFSET);
+            buffer.position(initPos + EosLogEntry.LOG_ENTRY_OVERHEAD);
```

Comment:
I feel `RECORDS_OFFSET` is more straight forward for reading at a given position?